src/Makefile.am

```diff
@@ -0,0 +1,48 @@
+# what flags you want to pass to the C compiler & linker
+CFLAGS = -Wall -g -O3 -DCAVEMAN_VERSION='"${VERSION}"' -DTEST_REF='"$(TEST_REF)"' -rdynamic
+
+# this lists the binaries to produce, the (non-PHONY, binary) targets in
+# the previous manual Makefile
+bin_PROGRAMS = caveman generateCavemanUMNormVCF
+caveman_SOURCES = caveman.c file_tests.c List.c List_algos.c bam_access.c config_file_access.c \
+                fai_access.c ignore_reg_access.c alg_bean.c split_access.c covs_access.c \
+                cn_access.c genotype.c algos.c output.c setup.c \
+                split.c mstep.c merge.c estep.c
+generateCavemanUMNormVCF_SOURCES = generateCavemanVCFUnmatchedNormalPanel.c file_tests.c List.c List_algos.c bam_access.c \
+                                config_file_access.c fai_access.c ignore_reg_access.c \
+                                alg_bean.c split_access.c covs_access.c cn_access.c \
+                                genotype.c algos.c output.c setup.c split.c \
+                                mstep.c merge.c estep.c
+
+caveman_LDADD=-lhts -lpthread -lz -lm -ldl -lrt
+caveman_CPPFLAGS=-I./
+generateCavemanUMNormVCF_LDADD=-lhts -lpthread -lz -lm -ldl -lrt
+generateCavemanUMNormVCF_CPPFLAGS=-I./
+
+# "make check" target
+TESTS = $(check_PROGRAMS)
+check_PROGRAMS = run_all_tests
+run_all_tests_SOURCES = ../tests/run_all_tests.c \
+        ../tests/check_alg_bean_tests.c \
+        ../tests/check_algos_tests.c \
+        ../tests/check_bam_access_tests.c \
+        ../tests/check_cn_access_tests.c \
+        ../tests/check_config_file_access_tests.c \
+        ../tests/check_covs_access_tests.c \
+        ../tests/check_fai_access_tests.c \
+        ../tests/check_genotype_tests.c \
+        ../tests/check_ign_region_access_tests.c \
+        ../tests/check_output_tests.c \
+        ../tests/check_split_access_tests.c \
+        file_tests.c List.c List_algos.c bam_access.c \
+        config_file_access.c fai_access.c ignore_reg_access.c \
+        alg_bean.c split_access.c covs_access.c cn_access.c \
+        genotype.c algos.c output.c setup.c split.c \
+        mstep.c merge.c estep.c
+run_all_tests_CFLAGS = -I../tests $(PTHREAD_CFLAGS)
+run_all_tests_LDADD = -lcheck -lhts -lpthread -lz -lm -ldl -lrt $(PTHREAD_LDFLAGS)
```

Comment:
While poking about [on internal FCE, bionic-WTSI-standard](https://zeta.internal.sanger.ac.uk/dashboard/project/instances/dd9f4870-1060-480b-bc4d-fd4cabdd78ca/), I found `make check` was getting a link error.  I fixed it with this bodge,

```diff
ubuntu@mca-caveytest2:~/gitwk-github/CaVEMan$ diff -u src/Makefile{~,}
--- src/Makefile~       2018-10-05 12:30:04.577995334 +0000
+++ src/Makefile        2018-10-05 15:23:36.989741543 +0000
@@ -588,7 +588,7 @@
         mstep.c merge.c estep.c
 
 run_all_tests_CFLAGS = -I../tests $(PTHREAD_CFLAGS)
-run_all_tests_LDADD = -lcheck -lhts -lpthread -lz -lm -ldl -lrt $(PTHREAD_LDFLAGS)
+run_all_tests_LDADD = -lcheck -lhts -lpthread -lz -lm -ldl -lrt $(PTHREAD_LDFLAGS) -lsubunit
 all: all-am
 
 .SUFFIXES:
```

and got

```
PASS: run_all_tests
============================================================================
Testsuite summary for CaVEMan 1.13.5
============================================================================
# TOTAL: 1
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

Previously,

```
ubuntu@mca-caveytest2:~/gitwk-github/CaVEMan$ make check
Making check in src
make[1]: Entering directory '/home/ubuntu/gitwk-github/CaVEMan/src'
make  run_all_tests
make[2]: Entering directory '/home/ubuntu/gitwk-github/CaVEMan/src'
gcc -I../tests -pthread -Wall -g -O3 -DCAVEMAN_VERSION='"1.13.5"' -DTEST_REF='""' -rdynamic  -L/home/ubuntu/gitwk-github/CaVEMan/hts_tmp -o run_all_tests ../tests/run_all_tests-run_all_tests.o ../tests/run_all_tests-check_alg_bean_tests.o ../tests/run_all_tests-check_algos_tests.o ../tests/run_all_tests-check_bam_access_tests.o ../tests/run_all_tests-check_cn_access_tests.o ../tests/run_all_tests-check_config_file_access_tests.o ../tests/run_all_tests-check_covs_access_tests.o ../tests/run_all_tests-check_fai_access_tests.o ../tests/run_all_tests-check_genotype_tests.o ../tests/run_all_tests-check_ign_region_access_tests.o ../tests/run_all_tests-check_output_tests.o ../tests/run_all_tests-check_split_access_tests.o run_all_tests-file_tests.o run_all_tests-List.o run_all_tests-List_algos.o run_all_tests-bam_access.o run_all_tests-config_file_access.o run_all_tests-fai_access.o run_all_tests-ignore_reg_access.o run_all_tests-alg_bean.o run_all_tests-split_access.o run_all_tests-covs_access.o run_all_tests-cn_access.o run_all_tests-genotype.o run_all_tests-algos.o run_all_tests-output.o run_all_tests-setup.o run_all_tests-split.o run_all_tests-mstep.o run_all_tests-merge.o run_all_tests-estep.o -lcheck -lhts -lpthread -lz -lm -ldl -lrt  -lz 
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libcheck.a(check_log.o): In function `subunit_lfun':
(.text+0x5f4): undefined reference to `subunit_test_start'
(.text+0x6bf): undefined reference to `subunit_test_fail'
(.text+0x6d4): undefined reference to `subunit_test_pass'
(.text+0x6ef): undefined reference to `subunit_test_error'
collect2: error: ld returned 1 exit status
Makefile:710: recipe for target 'run_all_tests' failed
make[2]: *** [run_all_tests] Error 1
make[2]: Leaving directory '/home/ubuntu/gitwk-github/CaVEMan/src'
Makefile:2048: recipe for target 'check-am' failed
make[1]: *** [check-am] Error 2
make[1]: Leaving directory '/home/ubuntu/gitwk-github/CaVEMan/src'
Makefile:378: recipe for target 'check-recursive' failed
make: *** [check-recursive] Error 1
```

Reply:
Logic added https://github.com/cancerit/CaVEMan/commit/4d7a655e5ad890a51d07d389041457e58ad088a9